src/main/java/org/testng/internal/MethodInvocationHelper.java

```diff
@@ -157,6 +157,8 @@ protected static Object invokeMethod(Method thisMethod, Object instance, Object[
         parameters.add(method);
       } else if (cls.equals(ITestContext.class)) {
         parameters.add(testContext);
+      } else if (cls.equals(Class.class)) {
+        parameters.add(com.getDeclaringClass());

```

Comment:
The idea was to cover every potential needs and Class is the only missing one.
Don't forget data providers may be an external class where we may need the test Class (but true, we may find it from the method too)